jans-cedarling/cedarling/src/common/policy_store.rs

```diff
@@ -5,34 +5,79 @@
  * Copyright (c) 2024, Gluu, Inc.
  */
 
-use std::collections::HashMap;
-
 use super::cedar_schema::CedarSchema;
+use serde::Deserialize;
+use std::collections::HashMap;
+use std::fmt;
 
 /// PolicyStoreMap it is a store for `PolicyStore` accessible by key.
-#[derive(Debug, serde::Deserialize)]
+#[derive(Debug, Deserialize)]
 pub(crate) struct PolicyStoreMap {
     #[serde(flatten)]
     pub policy_stores: HashMap<String, PolicyStore>,
 }
 
 /// PolicyStore contains all the data the Cedarling needs to verify JWT tokens and evaluate policies
-#[derive(Debug, Clone, serde::Deserialize)]
+#[derive(Debug, Clone, Deserialize)]
 pub struct PolicyStore {
     pub schema: CedarSchema,
     #[serde(deserialize_with = "deserialize_policy_set::parse")]
     pub policies: cedar_policy::PolicySet,
+    #[allow(dead_code)]
+    pub trusted_issuers: Option<Vec<TrustedIssuer>>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[allow(dead_code)]
+pub struct TrustedIssuer {
+    pub name: String,
+    pub description: String,
+    pub openid_configuration_endpoint: String,
+    pub token_metadata: Option<Vec<TokenMetadata>>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[allow(dead_code)]
+pub struct TokenMetadata {
+    #[serde(
+        rename = "type",
+        deserialize_with = "deserialize_policy_set::parse_token_kind"
+    )]
+    pub kind: TokenKind,
+    pub person_id: String, // the claim used to create the person entity
+    pub role_mapping: Option<String>, // the claim used to create a role for the token
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[allow(dead_code)]
```

Comment:
We can add "Copy" derive to the `TokenKind`, to avoid call copy method.
```suggestion
#[derive(Debug, Copy, Clone, Deserialize)]
```

Reply:
okay, i removed copy here: [98c50fb](https://github.com/JanssenProject/jans/pull/9893/commits/98c50fb6af5d6f36ef679fb203a94b6ed3f54562)